go/vt/mysqlctl/mysqld.go

```diff
@@ -1321,6 +1321,60 @@ func parseBinlogEntryTimestamp(logEntry string) (found bool, t time.Time, err er
 	return false, t, nil
 }
 
+// scanBinlogTimestamp invokes a `mysqlbinlog` binary to look for a timestamp in the given binary. The function
```

Comment:
mostly just refactored this function outside of `ReadBinlogFilesTimestamps`, but also changed a bit to ensure the termination of the `mysqlbinlog` process as I've found that in GitHub CI it might be slow to terminate and still holds on to the binlog file, which then conflicts with the next access to the file.